### PR TITLE
Batch backports to 6.0.x

### PR DESCRIPTION
**Pull request by jlucovsky**
Batch backports to 6.0.x

Link to [redmine](https://redmine.openinfosecfoundation.org/projects/suricata/issues) tickets:
- [6045](https://redmine.openinfosecfoundation.org/issues/6045)
- [6248](https://redmine.openinfosecfoundation.org/issues/6248)
- [6213](https://redmine.openinfosecfoundation.org/issues/6213)
- [6227](https://redmine.openinfosecfoundation.org/issues/6227)

Describe changes:
-
-
-

### Provide values to any of the below to override the defaults.

To use a pull request use a branch name like `pr/N` where `N` is the
pull request number.

Alternatively, `SV_BRANCH` may also be a link to an
OISF/suricata-verify pull-request.

```
SV_REPO=
SV_BRANCH=
SU_REPO=
SU_BRANCH=
LIBHTP_REPO=
LIBHTP_BRANCH=
```

### CODE REVIEW

File: src/suricata.c [hunk 1]
```diff
@@ -169,10 +172,31 @@
 
 #include "tmqh-packetpool.h"
 
-#include "util-proto-name.h"
+#include "util-byte.h"
```
Comment (victorjulien) [inline src/suricata.c:175]: are these large include changes really needed? Or are they a artifact of cherry-picking?

Comment (jlucovsky) [reply src/suricata.c:175]: From cherry-picking .... the harmful includes were removed but there are probably extra includes.

Comment (victorjulien) [reply src/suricata.c:175]: can you minimize it, we need backports to be as clean as possible

**Review by victorjulien — CHANGES_REQUESTED** [review]
needs include cleanups

**Comment by jlucovsky** [comment]
Continued in #9445

**Pull request closed without merging**